web/modules/custom/joinup_news/src/Guard/FulfillmentGuard.php

```diff
@@ -0,0 +1,219 @@
+<?php
+
+namespace Drupal\joinup_news\Guard;
+
+use Drupal\og\Og;
+use Drupal\state_machine\Guard\GuardInterface;
+use Drupal\state_machine\Plugin\Workflow\WorkflowInterface;
+use Drupal\state_machine\Plugin\Workflow\WorkflowTransition;
+use Drupal\Core\Entity\EntityInterface;
+
+/**
+ * Guard class for the transitions of the news entity.
+ *
+ * @package Drupal\joinup_news\Guard
+ */
+class FulfillmentGuard implements GuardInterface {
+  /**
+   * Constant representing the pre-moderating parent.
+   */
+  const MODERATED = 1;
+
+  /**
+   * Constant representing the post-moderating parent.
+   */
+  const NON_MODERATED = 0;
+
+  /**
+   * {@inheritdoc}
+   *
+   * We need to override default transitions allowed because this is also
+   * dependant on the parent's moderation, system roles and organic groups
+   * user roles. In the following method, the allowed transitions per
+   * moderation are checked and then if the transition is allowed, the user
+   * roles by the system and the organic groups are checked.
+   */
+  public function allowed(WorkflowTransition $transition, WorkflowInterface $workflow, EntityInterface $entity) {
+    $from_state = $entity->field_news_state->first()->value;
+    $parent = joinup_news_get_parent($entity);
+
+    $is_moderated = self::MODERATED;
+    if ($parent) {
+      $is_moderated = ($parent->bundle() == 'collection') ?
```

Comment:
These are two fields from different providers. One is og_group_ref and the other one is an entity_reference field, since collections are groups and solutions are not. I think there will be many problems it that happens.